src/modules/gateway/sagas.ts

```diff
@@ -248,38 +247,40 @@ export function* handleSetPurchase(action: SetPurchaseAction) {
     PurchaseStatus.FAILED,
     PurchaseStatus.CANCELLED
   ]
-  const { status, network, txHash } = purchase
+  const { status, network, txHash, nft } = purchase
 
-  if (
-    purchase.gateway === NetworkGatewayType.TRANSAK &&
-    purchase.status === PurchaseStatus.PENDING
-  ) {
-    window.addEventListener('beforeunload', handleBeforeUnload)
-  } else {
-    window.removeEventListener('beforeunload', handleBeforeUnload)
-  }
+  if (!nft) {
```

Comment:
Maybe this can be refactored to:

```
if (!nft) {
  yield call(handleManaSetPurchase, purchase)
}
```

(Pick a name for the handler that might be a better fit)